lib/pymedphys/tests/mosaiq/test_mock_patient_table.py

```diff
@@ -1,68 +1,174 @@
+from pymedphys._imports import numpy as np
 from pymedphys._imports import pandas as pd
 from pymedphys._imports import pymssql, pytest, sqlalchemy
 
-import pymedphys.mosaiq
+from pymedphys._mosaiq.helpers import get_patient_fields, get_patient_name
+from pymedphys.mosaiq import connect, execute
 
+msq_server = "."
+test_db_name = "MosaiqTest77008"
 
-@pytest.mark.mosaiqdb
-def test_mock_patient_table():
-    msq_server = "."
-    test_db_name = "MosaiqTest77008"
+sa_user = "sa"
+sa_password = "sqlServerPassw0rd"
+
+# set up a SQLAlchemy engine to be used for populating tables
+connection_str = f"mssql+pymssql://{sa_user}:{sa_password}@{msq_server}/{test_db_name}"
+engine = sqlalchemy.create_engine(connection_str, echo=False)
+
+
+@pytest.fixture(name="check_create_test_db")
+def fixture_check_create_test_db():
+    """ will create the test database, if it does not already exist on the instance """
+    # sa connection to create the test database
+    with pymssql.connect(
+        msq_server, user=sa_user, password=sa_password
+    ) as sql_sa_connection:
+
+        sql_sa_connection.autocommit(True)
 
-    sa_user = "sa"
-    sa_password = "sqlServerPassw0rd"
+        # create the test db
+        with sql_sa_connection.cursor() as cursor:
+            cursor.execute(
+                f"""
+                IF NOT EXISTS (SELECT * FROM sys.databases WHERE name = '{test_db_name}')
+                BEGIN
+                    CREATE DATABASE {test_db_name};
+                END
+                """
+            )
 
-    conn = pymssql.connect(msq_server, user=sa_user, password=sa_password)
-    conn.autocommit(True)
-    cursor = conn.cursor()
-    cursor.execute(f"create database {test_db_name}")
-    cursor.close()
 
-    patient_df = pd.DataFrame([("Larry"), ("Moe"), ("Curly")], columns=["FirstName"])
-    patient_df.index = patient_df.index + 10001  # use the index+10001 as the Pat_ID1
+@pytest.fixture(name="create_mock_patients")
+def fixture_create_mock_patients(check_create_test_db):
```

Comment:
pylint seems to complain that it doesn't understand what "check_create_test_db" is here -- not sure why...

Reply:
I think it was actually just saying that the variable wasn't being accessed within the function. I adjusted it a little to instead call the function within the fixture.